### DISCUSSION TITLE
Using Ariadne with newer versions of Django

### DISCUSSION THREAD

**Discussion by pkrakesh** [mirumee/ariadne#1148] [Q&A]
How to use Ariadne with newer versions of Django? I have no idea what to do since the 'ariadne-django' library currently is not maintained while the Ariadne documentation prescribes to use it, but no other details in this (Django) section.  Also the documentation on the home page of Ariadne is inconsistent with the ariadne-django documentation. 

#### Ariadne Doc
```
# ...or add GraphQL API to your favorite web framework
**from ariadne.contrib.django.views import GraphQLView**
from django.urls import include, path

from .schema import schema

urlpatterns = [
    ...
    path('graphql/', GraphQLView.as_view(schema=schema), name='graphql'),
]
```
#### ariadne-django Doc
```
**from ariadne_django.views import GraphQLView**
from django.urls import path

from .schema import schema

urlpatterns = [
    ...
    path('graphql/', GraphQLView.as_view(schema=schema), name='graphql'),
]
```
I am confused with what to do. I want to use Ariadne with Django 5, ASGI. More specifically: 1. Which libraries do I need to install to use Ariadne with Django now? 2. How to setup Ariadne with Dajngo ASGI? 3. How to write an Async resolver with newer versions of Django ASGI Application? 4. Do I need to use Django Channels as given in ariadne-django documentation to make my GraphQL API asynchronous?

```
from ariadne.asgi import GraphQL
from channels.http import AsgiHandler
from channels.routing import URLRouter
from django.urls import path, re_path


schema = ...


application = URLRouter([
    path("graphql/", GraphQL(schema, debug=True)),
    re_path(r"", AsgiHandler),
])
```

I am also using `ariadne-relay`. With Django 2.2 WSGI everything worked fine with ariadne-django. Now I need to make things asynchronous with Django's new support for ASGI out of the box. I moved from Graphene to Ariadne and absolutely love this library for the control it gives, schema-first approach and simplicity. Kindly help.

**Comment by pkrakesh** [comment]
Any points?

**Reply by rafalp** [reply]
It would be ideal if somebody maintained `ariadne-django`, but this is not the case.

**Comment by pkrakesh** [comment]
So changing to something like Strawberry is the best option when using Django?

**Reply by rafalp** [reply]
I guess so? Or contributing to `ariadne-django`.
